src/contrib/ad.jl

```diff
@@ -1,9 +1,9 @@
 const ADSUPPORT = (:ForwardDiff, :Zygote)
 const ADAVAILABLE = Dict{Module, Function}()
 
-Hamiltonian(metric::M, ℓπ::T, m::Module) where {M<:AbstractMetric,T} = ADAVAILABLE[m](metric, ℓπ)
+Hamiltonian(metric::AbstractMetric, ℓπ::Function, m::Module) = ADAVAILABLE[m](metric, ℓπ)
```

Comment:
Do we need this constraint to be a `Function`? This would prevent use of callables.

Reply:
It would be good to keep this more generic to support callables.

Reply:
I wasn’t aware of the callable use cases---will revert it.